spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/inbound/Mqttv5PahoMessageDrivenChannelAdapter.java

```diff
@@ -106,6 +110,28 @@ public Mqttv5PahoMessageDrivenChannelAdapter(MqttConnectionOptions connectionOpt
 		}
 	}
 
+	/**
+	 * Use this constructor when you need to use a single {@link ClientManager}
+	 * (for instance, to reuse an MQTT connection).
+	 * @param clientManager The client manager.
+	 * @param topic The topic(s).
+	 * @since 6.0
+	 */
+	public Mqttv5PahoMessageDrivenChannelAdapter(ClientManager<IMqttAsyncClient, MqttConnectionOptions> clientManager,
+			String... topic) {
+		super(clientManager, topic);
+		this.connectionOptions = clientManager.getConnectionInfo();
+	}
+
+	private static MqttConnectionOptions buildDefaultConnectionOptions(@Nullable String url) {
```

Comment:
I think we don't need this method any more.
Also I think we just need to add `Assert.hasText(url, "'url' cannot be null or empty");` into that url-based ctor.
Just because we cannot an empty url without external options: or url, or options with server uris.